packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -20,7 +21,7 @@ import 'shell.dart';
 
 abstract class AssetBundle {
   void close();
-  ImageResource loadImage(String key);
+  ImageResource loadImage(String key, [ AssetResolver resolver ]);
```

Comment:
I could create another subtree-context thingie like MediaQuery and use that to associate the asset resolver, but you might still want different resolution strategies in different widgets in the same build function, so I'm not sure that would actually work.